lib/container_orchestrator.rb

```diff
@@ -1,28 +1,84 @@
+require 'kubeclient'
+
 class ContainerOrchestrator
+  include_concern 'ObjectDefinition'
+
   TOKEN_FILE = "/run/secrets/kubernetes.io/serviceaccount/token".freeze
 
   def scale(deployment_config_name, replicas)
-    connection.patch_deployment_config(deployment_config_name, { :spec => { :replicas => replicas } }, ENV["MY_POD_NAMESPACE"])
+    connection.patch_deployment_config(deployment_config_name, { :spec => { :replicas => replicas } }, my_namespace)
+  end
+
+  def create_deployment_config(name)
+    definition = deployment_config_definition(name)
+    yield(definition) if block_given?
+    connection.create_deployment_config(definition)
+  rescue KubeException => e
+    raise unless e.message =~ /already exists/
```

Comment:
I wonder if we can open this as an RFE on KubeClient where they deal with the "already exists" message and raise a different exception.  Also, their exceptions should be namespaced (a second separate issue).

```ruby
def create_deployment_config(name)
  ...
rescue KubeClient::AlreadyExistsException
  # suppress
end